### ISSUE TITLE
[CORE] Downranking Formula Wrong

### ISSUE THREAD

**Issue by haephaistoss** [gamefreedomgit/Frostmourne#1264]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (https://www.wowhead.com/wotlk)

**Description:** Currently downranking doesn't work as intended.

**How to reproduce:** Cast any spell that is considered low rank - for max penalty use spells that's max caster level is 22 higher than actual caster level.

I used rank 1 frostbolt as test 1:

Results: 

Expected dmg: 21 - 24  (multipliers still work, but SP coeff has to be multiplied by 0)

Resulted dmg: 270 - 370 based on SP procs - these shouldn't have any effect on dmg.

Test 2:

Priest:

Flash Heal rank 1
expected range: 210 - 232
Result: 710 - 770

Some downranking clearly happens, that's why I suspect the formula may be an older pre 2.2 version.

**How it should work:**  From blue post (Ghostcrawler): "If caster level is less than max caster level, then the penalty = 1.0.
If caster level is at or greater than max caster level, then the penalty = (22 + max level - caster level) / 20.
The penalty is capped at 0. Max caster level per spell is also pretty inconsistent, but it's around 4 levels higher than the level at which you get the spell.
Basically, once the caster is 3 levels higher than the max cast level of the spell, the standard coefficient of the spell is multiplied by that penalty (losing 5% of the coefficient per level), with the coefficient reaching zero when the player is 22 levels higher than the max cast level."

**Database links:** https://wowpedia.fandom.com/wiki/Downranking

**Comment by Riztazz** [comment]
fixed, will close once it's live
